packages/grid/_modules_/grid/hooks/features/rows/useGridRows.ts

```diff
@@ -114,6 +114,36 @@ export const useGridRows = (
     [apiRef],
   );
 
+  const loadRows = React.useCallback(
+    (startIndex: number, pageSize: number, newRows: GridRowModel[]) => {
+      logger.debug(`Loading rows from index:${startIndex} to index:${startIndex + pageSize}`);
+
+      const newRowsToState = convertGridRowsPropToState(newRows, newRows.length, getRowIdProp);
+
+      setGridState((state) => {
+        const allRowsUpdated = state.rows.allRows.map((row, index) => {
+          if (index >= startIndex && index < startIndex + pageSize) {
+            return newRowsToState.allRows[index - startIndex];
+          }
+          return row;
+        });
+
+        internalRowsState.current = {
+          allRows: allRowsUpdated,
+          idRowsLookup: { ...state.rows.idRowsLookup, ...newRowsToState.idRowsLookup },
+          totalRowCount: state.rows.totalRowCount,
+        };
+
+        return { ...state, rows: internalRowsState.current };
+      });
```

Comment:
This section seems weird. Why not use the `apiRef.current.updateRows()` API? It supports addition.

Reply:
Maybe I didn't understand the `updateRows` but from what I can see it supports update, delete, and adding new rows but it can't replace existing rows, which is what I need here. but now thinking about it maybe it would be better to extend `updateRows` 🤔